garage/contrib/exp/agents/agent.py

```diff
@@ -0,0 +1,43 @@
+import abc
+
+
+class Agent(abc.ABC):
+    @abc.abstractmethod
+    def get_actions(self, obs):
+        """Get actions given a batch of observations.
+
+        Args:
+            obs: Tensor(batch_size, observation_dim)
+                 Observations.
+
+        Returns: Tensor(batch_size, action_dim)
+                 Actions.
+
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def train_once(self, samples):
+        """Train policy given trajectories sampled under latest policy.
+
+        Args:
+            samples: [Path]
+                Path: {
```

Comment:
Does this mean samples is a _collection of dictionaries of Tensors_, or a _dictionary of collections of Tensors_?

The distinction is important.

@krzentner do you think we should type these things with `akro`, or keep the types vanilla? There are upsides and downsides to both options.